bool right
bool up
bool left
bool back
bool radius
